packages/modal/src/positioning/ux-modal-positioning.ts

```diff
@@ -0,0 +1,409 @@
+    this.element.style.bottom = 'auto';
+    this.element.style.width = 'unset';
+    this.element.style.height = 'unset';
+  }
+
+  private getMainPlacement(): MainPlacement {
+    if (this.preferedPlacement === 'auto' || this.preferedPlacement === 'auto-end' || this.preferedPlacement === 'auto-start') {
+      return this.getAutoPlacement();
+    }
+    return this.preferedPlacement.split('-')[0] as 'left' | 'right' | 'top' | 'bottom';
+  }
+
+  private prepare() {
+    // set first variables that are needed in all scenarios
+    this.anchorRect = this.anchor.getBoundingClientRect();
+    this.elementRect = this.element.getBoundingClientRect();
+
+    this.hostOffsetX = 0;
+    this.hostOffsetY = 0;
+    if (this.element.parentElement) {
+      // set the host to relative if static
+      const style = window.getComputedStyle(this.element.parentElement);
+      if (style.position === 'static') {
+        this.element.parentElement.style.position = 'relative';
+      }
+      const rect = this.element.parentElement.getBoundingClientRect();
+      this.hostOffsetX = rect.left * -1;
+      this.hostOffsetY = rect.top * -1;
+      // If the host container has borders, they need to be offseted
+      // Important: this suppose a border-box box-sizing
+      // The reason is because the border is included in the sizing of the 
+      // host (width and height) but not included when it comes to positioning
+      // the child element (top:0, left:0 start inside the element)
+      if (style.borderLeftWidth) {
+        this.hostOffsetX = this.hostOffsetX - parseFloat(style.borderLeftWidth);
+      }
+      if (style.borderTopWidth) {
+        this.hostOffsetY = this.hostOffsetY - parseFloat(style.borderTopWidth);
+      }
+    }
+    this.constraintX = 0;
+    this.constraintY = 0;
+    this.constraintHeight = 0;
+    this.constraintWidth = window.innerWidth;
+    if (this.constraintElement instanceof HTMLElement) {
+      const rect = this.constraintElement.getBoundingClientRect();
+      this.constraintX = rect.left;
+      this.constraintY = rect.top;
+      this.constraintHeight = rect.height;
+      this.constraintWidth = rect.width;
+    }
+    this.spaceTop = undefined;
+    this.spaceRight = undefined;
+    this.spaceBottom = undefined;
+    this.spaceLeft = undefined;
+  }
+
+  private getSpaceTop() {
+    if (this.spaceTop !== undefined) {
+      return this.spaceTop;
+    }
+    this.spaceTop = this.anchorRect.top - this.constraintY - this.constraintMarginY;
+    return this.spaceTop;
+  }
+  
+  private getSpaceRight() {
+    if (this.spaceRight !== undefined) {
+      return this.spaceRight;
+    }
+    this.spaceRight = this.constraintElement === window ?
+      window.innerWidth - window.scrollX - this.anchorRect.width - this.anchorRect.left - this.constraintX - this.constraintMarginX :
+      this.constraintX + this.constraintWidth - this.anchorRect.width - this.anchorRect.left - this.constraintMarginX;
```

Comment:
Let's apply the same logic with coding style to ternary: relation ship of lines should be easy to skim, for that, let's put the `?` and `:` in the front:
```suggestion
    this.spaceRight = this.constraintElement === window
      ? window.innerWidth - window.scrollX - this.anchorRect.width - this.anchorRect.left - this.constraintX - this.constraintMarginX
      : this.constraintX + this.constraintWidth - this.anchorRect.width - this.anchorRect.left - this.constraintMarginX;
```